### PR TITLE
chore: Add nix shell for simple development setup

### PR DESCRIPTION
**Pull request by kylecarbs**
This enables contributors using Nix to set up their environment with ease.

Nix makes reproducibility at a project-level *really* nice. I'll be creating a dogfood template that uses the Nix shell, and hopefully we'll get the best of a few worlds :nerd_face:.

### CODE REVIEW

**Review by mafredri — COMMENTED** [review]
This is a very nice addition to the project! A few small tweaks and it should be ready to go. 👍🏻

File: scripts/build_go_slim.sh [hunk 1]
```diff
@@ -114,7 +114,7 @@ if [[ $compress != 0 ]]; then
 	sha_file=coder.sha1
 	sha_dest="$dest_dir/$sha_file"
 	log "--- Generating SHA1 for coder-slim binaries ($sha_dest)"
-	shasum -b -a 1 coder-* | tee $sha_file
+	openssl dgst -r -sha1 coder-* | cut -c1-40 | tee $sha_file
```
[the remark this answers is not in the document]
Comment (kylecarbs) [reply scripts/build_go_slim.sh:117]: Good catch!

File: .envrc [hunk 2]
```diff
@@ -0,0 +1 @@
+use nix
```
[the remark this answers is not in the document]
Comment (kylecarbs) [reply .envrc:1]: I agree with the override being odd. I'll just remove this for now, it doesn't seem standard anyways.

**Review by ammario — APPROVED** [review]
Docs LGTM

File: coderd/util/tz/tz_linux.go [hunk 3]
```diff
@@ -29,16 +29,16 @@ func TimezoneIANA() (*time.Location, error) {
...
 
-	lp, err := filepath.EvalSymlinks(etcLocaltime)
-	if err != nil {
-		return nil, xerrors.Errorf("read location of %s: %w", etcLocaltime, err)
+	location, err := filepath.EvalSymlinks(etcLocaltime)
+	if err == nil {
+		location = strings.Replace(location, zoneInfoPath, "", -1)
+		location = strings.TrimPrefix(location, string(filepath.Separator))
+	} else {
+		location, _ = time.Now().Zone()
```
[the remark this answers is not in the document]
Comment (kylecarbs) [reply coderd/util/tz/tz_linux.go:37]: Looking through the source, it seems that our `TimezoneIANA` function duplicates Go stdlib `time.Local`. Could we just use that instead?

[the remark this answers is not in the document]
Comment (mafredri) [reply coderd/util/tz/tz_linux.go:37]: In what distro/env are you running into the issue? I'd think that if neither `TZ` nor `/etc/localtime` is set, Go wouldn't know what timezone to use either, at which point defaulting to `UTC` (e.g. `time.LoadLocation("")` or `time.LoadLocation("UTC")`) would be a valid approach?

Another approach could be to:

```go
name, offset := time.Now().Zone()
loc := time.FixedZone(name, offset)
```

However, **we might break some stuff**. We're essentially creating a new timezone with a custom name (e.g. `EEST`).

**Review by mafredri — APPROVED** [review]
Looks great! 💪🏻

**Pull request merged**